sympy/integrals/integrals.py

```diff
@@ -516,6 +517,48 @@ def try_meijerg(function, xab):
                             function = ret
                             continue
 
+            if not isinstance(antideriv,Integral) and antideriv is not None:
+                sym = xab[0]
+                if antideriv.is_Add:
+                    terms = antideriv.args
+                else:
+                    terms = tuple((antideriv,))
+                for term in terms:
+                    atan_arg = term.atoms(atan)
```

Comment:
Would it be possible to search `atan` nodes in `antideriv` instead of each one of its arguments separately?

Reply:
We can search for `atan` nodes in each one argument, and then have to find limit of that `term` (argument).
Otherwise if we search in `antideriv`, then we don't know in which term `atan` is present and we won't be able to apply limit correctly.

Reply:
Should it then be checked that each argument of `Add` would contain at most one `atan` node?

Reply:
Yes. Because if more than one `atan` nodes were present then `floor` shouldn't be added or some another special term will be added!

Reply:
If there is only one `atan` node in an argument, then the `floor` term can be added to that directly. There should be no need to compute limits as we know the limits of `atan`. (Replace the `atan` node by an `Add` node with two arguments, `atan` and `floor`.)

Reply:
yes. That is a better solution to replace `atan` by an `Add` node having two arguments `atan` and `floor` !

Reply:
And the term should be checked for that it contains only one `atan` node ?

Reply:
In principle, it may be possible to have several `atan` terms, if there will be no confusion.